src/pages/VenueLandingPage/VenueLandingPage.tsx

```diff
@@ -91,6 +104,56 @@ export const VenueLandingPage: React.FunctionComponent<VenueLandingPageProps> =
     }
   }, [shouldOpenPaymentModal, isAuthenticationModalOpen]);
 
+  const onJoinClick = useCallback(() => {
+    if (venueId) {
+      const venueEntrance = venue?.entrance && venue.entrance.length;
+      window.location.href =
+        user && !venueEntrance
+          ? venueInsideUrl(venueId)
+          : venueEntranceUrl(venueId);
+    }
+  }, [user, venue, venueId]);
+
+  const renderJoinButton = useCallback(() => {
+    switch (venue?.accessMode) {
+      case VenueAccessMode.password:
+        return (
+          <div className="secret-password-form-wrapper">
+            <SecretPasswordForm
+              buttonText={
+                venue.config?.landingPageConfig.joinButtonText ??
+                "Join the party"
+              }
+              onSuccess={showJoinVenueModal}
+            />
+          </div>
+        );
+
+      case VenueAccessMode.email:
+      case VenueAccessMode.ticket:
+      case VenueAccessMode.parent:
+      case VenueAccessMode.open:
+      default:
+        return (
+          <button
+            className="btn btn-primary btn-block btn-centered"
+            onClick={onJoinClick}
+          >
+            Join the event
+            {(venue?.start_utc_seconds ?? 0) > new Date().getTime() / 1000 && (
+              <span className="countdown">
+                Begins in {getTimeBeforeParty(venue?.start_utc_seconds)}
+              </span>
+            )}
+          </button>
+        );
+    }
+  }, [onJoinClick, showJoinVenueModal, venue]);
+
+  const hideJoinVenueModal = useCallback(() => {
+    setIsJoinVenueModalOpen(false);
+  }, []);
```

Comment:
I would tend to put these alongside their corresponding `useState` to keep everything 'grouped together'